browser/ui/webui/brave_rewards_page_ui.cc

```diff
@@ -568,6 +584,19 @@ void RewardsDOMHandler::GetAutoContributeProperties(
                      weak_factory_.GetWeakPtr()));
 }
 
+void RewardsDOMHandler::SetExternalWalletType(const base::ListValue* args) {
+  CHECK_EQ(1U, args->GetSize());
+  if (!rewards_service_)
+    return;
+
+  AllowJavascript();
+  const std::string wallet_type = args->GetList()[0].GetString();
+  rewards_service_->SetExternalWalletType(wallet_type);
+  rewards_service_->GetExternalWallet(
```

Comment:
we shouldn't be calling `GetExternalWallet` here. If the page wants the external wallet, it should call it directly, but in this case what you want is a pref registrar that fires whenever the pref is updated from anywhere

Reply:
resolved: https://github.com/brave/brave-core/pull/8566/files#diff-80b4a43a6852eec511431cc9664e5474ac9626ab0ad84bd55f3b0b7f5d9c5dedR510